src/index.js

```diff
@@ -518,10 +518,12 @@ function createConfig(options, entry, format, writeMeta) {
 							tsconfigDefaults: {
 								compilerOptions: {
 									sourceMap: options.sourcemap,
-									declaration: true,
+									declaration: options.generateTypes !== false,
 									allowJs: true,
 									emitDeclarationOnly: options.generateTypes && !useTypescript,
-									declarationDir: getDeclarationDir({ options, pkg }),
+									...(options.generateTypes !== false && {
+										declarationDir: getDeclarationDir({ options, pkg }),
+									}),
```

Comment:
The behaviour here is the same as if we had simply set `declarationDir` to `undefined`, but rpt2 throws a bunch of errors  if that key exists at all & `declaration` is `false`.

```
$ microbundle --generateTypes false
rpt2: options error TS5069: Option 'declarationDir' cannot be specified without specifying option 'declaration' or option 'composite'.
rpt2: options error TS5069: Option 'declarationDir' cannot be specified without specifying option 'declaration' or option 'composite'.
rpt2: options error TS5069: Option 'declarationDir' cannot be specified without specifying option 'declaration' or option 'composite'.
rpt2: options error TS5069: Option 'declarationDir' cannot be specified without specifying option 'declaration' or option 'composite'.
```